presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -46,11 +52,24 @@ public IterativeOptimizer(StatsRecorder stats, Set<Rule> rules)
 
     public IterativeOptimizer(StatsRecorder stats, List<PlanOptimizer> legacyRules, Set<Rule> newRules)
     {
+        this(stats, legacyRules, newRules, 5000);
```

Comment:
I don't remember such a comment. But I agree with it completely.
Will provide a property to configure it. In case it's not specified, timeout is not enforced.